File: src/main/java/com/nurkiewicz/tsclass/parser/ast/expr/Identifier.java
package com.nurkiewicz.tsclass.parser.ast.expr;

import lombok.Value;

@Value
public class Identifier implements Expression {

    private final String name;

    public static Identifier ident(String name) {
        return new Identifier(name);
    }

    public String toString() {
        return name;
    }

}

File: src/main/kotlin/com/nurkiewicz/tsclass/parser/ast/expr/Identifier.kt
package com.nurkiewicz.tsclass.parser.ast.expr

data class Identifier(val name: String) : Expression {

    override fun toString(): String {
        return name
    }

    companion object {

        @JvmStatic
        fun ident(name: String): Identifier {
            return Identifier(name)
        }
    }

}
